src/pages/index.js

```diff
@@ -62,36 +104,160 @@ function handleImageClick(cardData) {
   };
-  const cardElement = createCard(card);
-  section.addItems(cardElement);
-  addcardPopup.close();
-  profileAddFormElement.reset();
-  addFormValidator.disableButton();
+  renderLoading(true, addCardPopup);
+  // Create the Card 
+  api.addNewCard(cardData)
+    .then((newCardData) => {
+      const cardElement = createCard(newCardData);
+      section.addItems(cardElement);  
+      addCardPopup.close();            
+      profileAddFormElement.reset();  
+      addFormValidator.disableButton();  
+    })
+    .catch(err => {
+      console.error(`Error adding card: ${err}`);
+    })
+    .finally(() => {
+      renderLoading(false, addCardPopup)
+    }
+    )
 }
 
-function openEditProfileModal() {
-  const currentUser = userInfo.getUserInfo();
+
+
+//avatar
+function handleAvatarSubmit(data){
+  renderLoading(true, avatarPopup);
+  api.updateUserAvatar({
+    avatar: data.url
+  })
+    .then(updateUserData => {
+      user.setUserInfo({
+        avatar: updateUserData.avatar
+      });
+      
+    })
+    .catch(err => console.error(`Error updating avatar: ${err}`))
+    .finally(() =>{
+      renderLoading(false, avatarPopup);
+      avatarPopup.close();
+      avatarFormElement.reset();
+    });
+}
+
+//delete
+function handleDeleteSubmit(card){
+  const confirmButton = document.querySelector('#delete-confirm-button');
+  deletePopup.open();
+  
+  confirmButton.addEventListener('click', function onConfirmClick() {
+    renderLoading(true, deletePopup);
+    card.removeCard();
+    api.deleteCard(card._id)
+      .then(card => {
+          
+          confirmButton.removeEventListener('click', onConfirmClick)
+      })
+      .catch(err => {
+        console.error(`Error deleting card ${err}`);
+      })
+      .finally(()=>{
+        renderLoading(false, deletePopup);
+        deletePopup.close();
+      })
+  })
+}
+  
+    
+    
+  
+
+//like
+function handleCardLike(card){
+  const isLiked = card.isLiked(); 
+
+  const likePromise = isLiked
+    ? api.removeLike(card.getId())  
+    : api.addLike(card.getId()); 
+
+  likePromise
+    .then(updatedCardData => {
+      card.updateLikes(updatedCardData.isLiked);  
+    })
+    .catch(err => console.error(`Error updating like status: ${err}`));
+}
+
+function openModal(popup) {
+  const currentUser = user.getUserInfo();
   userNameInput.value = currentUser.name;
-  userJobInput.value = currentUser.job;
-  editPopup.open();
+  userJobInput.value = currentUser.about;
+  popup.open();
 }
 
+function renderLoading(isSaving, popup){
+  if(isSaving){
+    if(popup == deletePopup){
+      return popup.confirmButton.textContent = "Deleating Card";
+    }
+    if(popup == addCardPopup){
+      return popup.submitButton.textContent = "Creating Card";
+    }
+    if (popup == editPopup || avatarPopup){
+      return popup.submitButton.textContent = "Saving";
+    }
+    
+        
+  }
+  else{
+    if(popup == deletePopup){
+      return popup.confirmButton.textContent = "Yes";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```

Reply:
Added this feature in  Added Renderloading to both PopupForm and PopupWithConfirm Classes instead of index (https://github.com/Schou10/se_project_aroundtheus/pull/1/commits/e77aa0837b70af26e0f9c6ea73663bb35a87aebd)